### PR TITLE
changes according to TaskItem instead of simple string

### PR DESCRIPTION
**Pull request by bodik4real**
-added Task Item and integrate it in code

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -13,16 +13,22 @@ import { TaskListGuard } from './task-list.guard';
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
 
+import { ModalModule } from 'ngx-bootstrap';
+
+
```
Comment (timonsk) [inline src/app/app.module.ts]: Remove extra empty line

File: src/app/app.module.ts [hunk 2]
```diff
@@ -13,16 +13,22 @@ import { TaskListGuard } from './task-list.guard';
...
     LoginModule,
     TaskListModule,
+    
     FormsModule,
     AppRoutingModule,
     ReactiveFormsModule,
-    BrowserAnimationsModule
+    BrowserAnimationsModule,
+    ModalModule.forRoot()
+
```
Comment (timonsk) [inline src/app/app.module.ts]: Same here

File: src/app/models/TaskItem.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export class Task {
+export class TaskItem {
     public id: number;
     public value: string
```
Comment (timonsk) [inline src/app/models/TaskItem.ts]: Please add ; and }

File: src/app/services/user-data.service.ts [hunk 4]
```diff
@@ -61,4 +65,9 @@ export class UserDataService {
   public redirectToLogin() {
     this.router.navigate(['/login']);
   }
+  
+  public saveUser(user: User)
+  {
```
Comment (timonsk) [inline src/app/services/user-data.service.ts]: Please format this file as appropriate

File: src/app/task-list/task-list-item/task-list-item.component.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+@Component({
+  selector: 'app-task-list-item',
+  templateUrl: './task-list-item.component.html',
+  styleUrls: ['./task-list-item.component.css']
+})
+export class TaskListItemComponent {
+
+  public isTaskEditing: boolean;
+
+  constructor() {}
```
Comment (timonsk) [inline src/app/task-list/task-list-item/task-list-item.component.ts]: Remove useless constructor

File: src/app/task-list/task-list.component.ts [hunk 6]
```diff
@@ -23,26 +25,27 @@ export class TaskListComponent {
...
-  {
-    this.taskDataService.deleteTask(task);
+  public deleteTask(taskIndex: number): void {
+    this.taskDataService.deleteTask(taskIndex);
     for (let i = 0; i < this.tasks.length; i++) {
-      if (this.tasks[i] == task) {
+      if (this.tasks[i].id === taskIndex) {
         this.tasks.splice(i, 1);
       }
+     // this.modalRef.hide();
```
Comment (timonsk) [inline src/app/task-list/task-list.component.ts]: Remove useless comments

File: src/app/task-list/task-list.component.ts [hunk 7]
```diff
@@ -23,26 +25,27 @@ export class TaskListComponent {
...
   }
 
-  public deleteTask(task: string):void
-  {
-    this.taskDataService.deleteTask(task);
+  public deleteTask(taskIndex: number): void {
+    this.taskDataService.deleteTask(taskIndex);
     for (let i = 0; i < this.tasks.length; i++) {
-      if (this.tasks[i] == task) {
+      if (this.tasks[i].id === taskIndex) {
```
Comment (timonsk) [inline src/app/task-list/task-list.component.ts]: We can find index as we did in editTask below

File: src/app/task-list/task-list.component.html [hunk 8]
```diff
@@ -2,7 +2,7 @@
   <div class="col-lg-3 col-lg-offset-4">
     <form [formGroup]="taskForm" (submit)="addTask(task.value)">
       <div class="input-group">
-        <input type="text" name="task" formControlName="task" placeholder="typeYourTaskHere" class="form-control" #task>
+        <input type="text" name="task" formControlName="task" placeholder="Type Your Task Here" class="form-control" #task>
```
Comment (timonsk) [inline src/app/task-list/task-list.component.html]: Please use lowercase for all except the first word

File: src/app/services/user-data.service.ts [hunk 9]
```diff
@@ -4,19 +4,22 @@ import { User } from '../models/User';
...
 
 @Injectable({
   providedIn: 'root'
 })
 export class UserDataService {
 
-  testTaskList: string[];
+  private currentUserKey = 'currentUser';
+  private dateFormat = "DD-MM-YYYY";
+  testTaskList: Array<TaskItem>;
```
Comment (timonsk) [inline src/app/services/user-data.service.ts]: Please add accesses modifier here like public or private

Comment (bodik4real) [reply src/app/services/user-data.service.ts]: Fix is done.

File: src/app/task-list/task-list.component.html (hunk 8 above)
Comment (bodik4real) [reply src/app/task-list/task-list.component.html]: Fix is done.

File: src/app/task-list/task-list.component.ts (hunk 7 above)
Comment (bodik4real) [reply src/app/task-list/task-list.component.ts]: Fix is done.

File: src/app/task-list/task-list.component.ts (hunk 6 above)
Comment (bodik4real) [reply src/app/task-list/task-list.component.ts]: I suppose, I’ll need this line in nearest future.

File: src/app/task-list/task-list-item/task-list-item.component.ts (hunk 5 above)
Comment (bodik4real) [reply src/app/task-list/task-list-item/task-list-item.component.ts]: Fix is done.

File: src/app/services/user-data.service.ts (hunk 4 above)
Comment (bodik4real) [reply src/app/services/user-data.service.ts]: Fix is done.

File: src/app/models/TaskItem.ts (hunk 3 above)
Comment (bodik4real) [reply src/app/models/TaskItem.ts]: Fix is done.

File: src/app/app.module.ts (hunk 2 above)
Comment (bodik4real) [reply src/app/app.module.ts]: Fix is done.

File: src/app/app.module.ts (hunk 1 above)
Comment (bodik4real) [reply src/app/app.module.ts]: Fix is done.

File: src/app/services/user-data.service.ts [hunk 10]
```diff
@@ -9,17 +9,18 @@ import { TaskItem } from '../models/TaskItem';
...
   private dateFormat = "DD-MM-YYYY";
-  testTaskList: Array<TaskItem>;
+  private testTaskList: Array<TaskItem>;
 
   constructor(private router: Router) {
-     this.testTaskList = [new TaskItem(0,"fewfw")];
-     var date = moment("2013-03-24")
-     var testUser = new User("TestName", "TestSurName", "password", date.toString(), this.testTaskList);
-     localStorage.setItem(this.currentUserKey, JSON.stringify(testUser));
+    this.testTaskList = [new TaskItem(0, "fewfw")];
```
Comment (timonsk) [inline src/app/services/user-data.service.ts]: Please move this to some method and call it from the constructor

Comment (bodik4real) [reply src/app/services/user-data.service.ts]: fix is done

**Pull request merged**